WIP: Log Group events

frisi opened the pull request:
this logs deletion of PAS Groups too.

requires recent versions of the following packages  (both shipped with plone >= 4.3.8)

* Products.PluggableAuthService 1.11.0
  https://github.com/zopefoundation/Products.PluggableAuthService/pull/3

* PlonePAS 5.0.8
  https://github.com/plone/Products.PlonePAS/commit/c44e655f782b8a978862b1575658a8aad5b30f2f


In the future there will be separate events for users and groups (see [the discussion in the pull request for pluggableauthservice](https://github.com/zopefoundation/Products.PluggableAuthService/pull/3) and [a commit by that will probably end up in Plone](https://github.com/IMIO/Products.PlonePAS/commit/2c67baeae19bf0d05722f23179d0a9f06db9efcb), correct @DieKatze?

i've got two issues with this PR:

### A) it does not work with plone < 4.3.8

i need help for a proper zcml condition so we can only register the subscriber if the `IGroupDeletedEvent` is available.

can be tested by changing `buildout.cfg` to

```
extends =
    https://raw.github.com/collective/buildout.plonetest/master/test-4.3.x.cfg
    http://dist.plone.org/release/4.3.7/versions.cfg
    https://raw.github.com/collective/buildout.plonetest/master/qa.cfg
    https://raw.github.com/plone/plone.app.robotframework/master/versions.cfg
```

### B) the check for the import in pas_plugins

the import test via `import pkg_resources` [as suggested by the styleguide](http://docs.plone.org/4/en/external/plone.api/docs/contribute/conventions.html#about-imports)
does not work for `Products.PluggableAuthService.interfaces.events.IGroupDeletedEvent`
we'd need to check for the version of PluggableAuthService or leave it as is:

```
try:
    from Products.PluggableAuthService.interfaces.events import IGroupDeletedEvent
    HAS_GROUPEVENTS = True
except ImportError:
    HAS_GROUPEVENTS = False
```

frisi commented:
@pcdummy fyi

hvelarde commented:
pinning both packages in setup.py should be enough:

```
Products.PluggableAuthService >= 1.11.0
PlonePAS >= 5.0.8
```

a rebase is needed because you created the branch from the one with the German translation.

hvelarde commented:
@frisi this breaks compatibility with Plone 4.2 and I'm fine with that after releasing a working version of this package.

if you agree, I will release 1.0b2 (compatible with Plone 4.2) and the we can start working on 1.1b1 (drop compatibility with Plone 4.2).

frisi commented:
@hvelarde i'm perfectly fine with releasing a version w/o this feature. so please go ahead with the 1.0b2 release

i don't think that pinning the versions for the PAS components in setup.py is the way to go.

maybe @mauritsvanrees, @gforcada @vangheem or @jensens have an idea how to do a proper zcml-condition for our usecase (see description of this PR)

hvelarde commented:
@frisi if a newer version does not contain that feature, then we have an issue outside this package.

release is done: https://pypi.python.org/pypi/collective.fingerpointing/1.0b2

thanks!

frisi commented:
> @frisi if a newer version does not contain that feature, then we have an issue outside this package.

imagine 1.x series is used for plone4.3.x (and 1.1.5 introduces the feature) and 2.x for plone 5.0.x (and the feature has been added in 2.1.3). in this case on a plone5 site that uses 2.1.2 the requirement (version >=1.1.5) will be met. however, the site will be broken.

hvelarde commented:
@frisi sorry, but I'm not following you: 1.x and 2.x series of what? this package is already compatible with both, Plone 4.x and 5.x.

hvelarde commented:
for me there's no problem with this: Plone 4.3.8 is a bug fix release; anybody running any previous version of 4.3 branch can easily update versions of both packages without any issue.

please remove the conditionals as they are no longer needed and add a test for this new feature.

jensens commented on setup.py:
```diff
@@ -45,9 +44,10 @@
         'plone.registry',
         'Products.CMFCore',
-        'Products.CMFPlone >=4.2',
+        'Products.CMFPlone >=4.3',
```
so if i understand correctly this should be changed to >=4.3.8 ?

mauritsvanrees replied:
Pinning PlonePAS/PluggableAuthService seems sufficient. Then someone can choose to use say CMFPlone 4.3.0 and change the default pins for the PAS packages.

hvelarde replied:
exactly!

frisi commented:
thanks for your feedback @hvelarde , @mauritsvanrees and @jensens 

rebased, removed the conditions and added a test.

ready for review @hvelarde.

frisi commented:
@hvelarde if everything is fine, a new release would be great, too ;-)

hvelarde commented:
@frisi I'l make the last release compatible with Plone 4.2; after that I'll merge and make another one dropping the support.

frisi commented:
fine with that @hvelarde  ;-)
i'd say it's fine to change the version number on a branch to indicate that the next release should be a new feature release as you did in https://github.com/collective/collective.fingerpointing/pull/28/commits/a841d99e4952a4b99c0b33c8dedacd1f3bc91ae5

why do you think it's better to [revert that](https://github.com/collective/collective.fingerpointing/pull/28/commits/d75f1603210c2adf46866ec940aa00cb3cc576c9)?

hvelarde commented:
@frisi yes, I do that sometimes because I'm the one making the releases ;-)
